content/en/blog/_posts/2019-12-09-volume-snapshot-beta.md

```diff
@@ -0,0 +1,449 @@
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshotContent
+metadata:
+  name: manually-created-snapshot-content
+spec:
+  deletionPolicy: Delete
+  driver: testdriver.csi.k8s.io
+  source:
+    snapshotHandle: 7bdd0de3-aaeb-11e8-9aae-0242ac110002
+  volumeSnapshotRef:
+    name: test-snapshot
+    namespace: default
+```
+
+Once a `VolumeSnapshotContent` object is created, a user can create a `VolumeSnapshot` object pointing to the `VolumeSnapshotContent` object. The name and namespace of the `VolumeSnapshot` object must match the name/namespace specified in the volumeSnapshotRef of the VolumeSnapshotContent. It specifies the following fields:
+`volumeSnapshotContentName` - name of the volume snapshot content specified above. This field is required.
+`volumeSnapshotClassName` - name of the volume snapshot class. This field is optional.
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshot
+metadata:
+  name: manually-created-snapshot
+spec:
+  source:
+        volumeSnapshotContentName: test-content
+```
+
+Once both objects are created, the common snapshot controller verifies the binding between `VolumeSnapshot` and `VolumeSnapshotContent` objects is correct and marks the `VolumeSnapshot` as ready (if the CSI driver supports the `ListSnapshots` call, the controller also validates that the referenced snapshot exists). The CSI external-snapshotter sidecar checks if the snapshot exists if ListSnapshots CSI method is implemented, otherwise it assumes the snapshot exists. The external-snapshotter sidecar sets `readyToUse` to true in the status field of `VolumeSnapshotContent`. The common snapshot controller marks the snapshot as ready accordingly.
+Create Volume From Snapshot
+Once you have a bound and ready `VolumeSnapshot` object, you can use that object to provision a new volume that is pre-populated with data from the snapshot.
+
+To provision a new volume pre-populated with data from a snapshot, use the `dataSource` field in the `PersistentVolumeClaim`. It has three parameters:
+`name` - name of the `VolumeSnapshot` object representing the snapshot to use as source
+`kind` - must be `VolumeSnapshot`
+`apiGroup` - must be snapshot.storage.k8s.io
+
+The namespace of the source `VolumeSnapshot` object is assumed to be the same as the namespace of the `PersistentVolumeClaim` object.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-restore
+  namespace: demo-namespace
+spec:
+  storageClassName: testdriver.csi.k8s.io
+  dataSource:
+    name: manually-created-snapshot
+    kind: VolumeSnapshot
+    apiGroup: snapshot.storage.k8s.io
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+When the `PersistentVolumeClaim` object is created, it will trigger provisioning of a new volume that is pre-populated with data from the specified snapshot.
```

Comment:
nit:
```suggestion
When the PersistentVolumeClaim object is created, it will trigger provisioning of a new volume that is pre-populated with data from the specified snapshot.
```